### PR TITLE
Fixes in Uri and Curl Client

### PR DESCRIPTION
**Pull request by midahp**
- fix Uri::__toString
- fix Curl Client headers and body

### CODE REVIEW

File: src/Client/Curl.php [hunk 1]
```diff
@@ -26,14 +30,16 @@
 use Horde\Http\Response;
 use Horde\Http\ResponseFactory;
 use Horde\Http\StreamFactory;
-use \Horde_Support_CaseInsensitiveArray;
+use Horde_Support_CaseInsensitiveArray;
```
Comment (ralflang) [inline src/Client/Curl.php]: Please use the namespaced version of Horde\Support in namespaced code.

File: src/Client/Fopen.php [hunk 2]
```diff
@@ -110,7 +112,7 @@ public function sendRequest(RequestInterface $request): ResponseInterface
         // Stream context config.
         $opts['http']['method'] = $method;
         $opts['http']['header'] = implode("\n", $hdr);
-        $opts['http']['content'] = $data;
+        $opts['http']['content'] = (string) $data;
```
Comment (ralflang) [inline src/Client/Fopen.php:115]: Shouldn't we try to keep the original resource (if it is a resource) as long a spossible and only cast to string / open a new resource if needed? Casting to string may be a major memory burden for larger up/downloads.

File: src/Uri.php [hunk 3]
```diff
@@ -1,13 +1,17 @@
 <?php
+
 declare(strict_types=1);
+
 namespace Horde\Http;
-use \Horde_String;
+
+use Horde_String;
```
Comment (ralflang) [inline src/Uri.php]: Please use Horde\Util\HordeString;

**Review by ralflang — CHANGES_REQUESTED** [review]
Mostly fine, some minor details to change first.

File: src/Client/Fopen.php (hunk 2 above)
Comment (midahp) [reply src/Client/Fopen.php:115]: Makes sense, but the documentation says this needs to be of type string: https://www.php.net/manual/en/context.http.php#context.http.content.
I tested it and it didn't work without casting it to string here.

**Pull request merged**